doc: Update manuals bespoke syntax (#264450)
NixOS/nixpkgs: doc/README.md

watermelon
:   green fruit with red flesh
```

#### Examples

To define a referenceable figure use the following fencing:

```markdown
:::{.example #an-attribute-set-example}
# An attribute set example

You can add text before

    ```nix
    { a = 1; b = 2;}
    ```

and after code fencing
:::
```

> Defining examples through the `example` fencing class adds them to a "List of Examples" section after the Table of Contents.

Reviewer: ```suggestion
Defining examples through the `example` fencing class adds them to a "List of Examples" section after the Table of Contents.
Though this is not shown in the rendered documentation on nixos.org.
```

#### Figures

To define a referencable figure use the following fencing:

```markdown
::: {.figure #nixos-logo}
# NixOS Logo
![NixOS logo](./nixos_logo.png)
:::
```

> Defining figures through the `figure` fencing class adds them to a `List  of Figures` after the `Table of Contents`.

Reviewer: ```suggestion
Defining figures through the `figure` fencing class adds them to a `List  of Figures` after the `Table of Contents`.
Though this is not shown in the rendered documentation on nixos.org.
```

#### Footnotes

To add a foonote explanation, use the following syntax:

```markdown
Sometimes it's better to add context [^context] in a footnote.

[^context]: This explanation will be rendered at the end of the chapter.
```

#### Inline comments

Inline comments are supported with following syntax:

```markdown
<!-- This is an inline comment -->
```

The comments will not be rendered in the rendered HTML.

#### Link reference definitions

Links can reference a label, for example, to make the link target reusable:

```markdown
::: {.note}

> Link referencess can also be used to [shorten URLs][url-id] and keep the markdown readable.

Reviewer 2: ```suggestion
Reference links can also be used to [shorten URLs][url-id] and keep the markdown readable.
```

:::

[url-id]: https://github.com/NixOS/nixpkgs/blob/19d4f7dc485f74109bd66ef74231285ff797a823/doc/README.md
```

This syntax is taken from [CommonMark](https://spec.commonmark.org/0.30/#link-reference-definitions).

#### Typographic replacements

> Typographic replacements are enabled. For a list of  possible replacement examples check [here](https://github.com/executablebooks/markdown-it-py/blob/3613e8016ecafe21709471ee0032a90a4157c2d1/markdown_it/rules_core/replacements.py#L1-L15).

Reviewer 2: ```suggestion
Typographic replacements are enabled. Check the [list of possible replacement patterns check](https://github.com/executablebooks/markdown-it-py/blob/3613e8016ecafe21709471ee0032a90a4157c2d1/markdown_it/rules_core/replacements.py#L1-L15).
```

watermelon
:   green fruit with red flesh
```

#### Examples

To define a referenceable figure use the following fencing:

```markdown
:::{.example #an-attribute-set-example}
# An attribute set example

You can add text before

>     ```nix
>     { a = 1; b = 2;}
>     ```

Reviewer 2: ````suggestion
 ```nix
{ a = 1; b = 2;}
```
````

Author: This breaks the rendering on Github: https://github.com/NixOS/nixpkgs/pull/264450/commits/c9e40048554fd59e0a3a5af8939dfd85a8a03744

Reviewer 2: Ah you need four (strictly more than three) backticks in the surrounding code block!

and after code fencing
:::
```

> Defining examples through the `example` fencing class adds them to a `List  of Examples` after the `Table of Contents`.

Reviewer 2: ```suggestion
Defining examples through the `example` fencing class adds them to a "List of Examples" section after the Table of Contents.
```

#### Figures

To define a referencable figure use the following fencing:

```markdown
::: {.figure #nixos-logo}
# NixOS Logo
![NixOS logo](./nixos_logo.png)
:::
```

Defining figures through the `figure` fencing class adds them to a `List  of Figures` after the `Table of Contents`.

#### Footnotes

Footnotes are supported as a URL shortener alternative:

```markdown
::: {.note}
Footnotes can be used to [shorten URLs][url-id] and keep the markdown readable.
:::

[url-id]: https://github.com/NixOS/nixpkgs/blob/19d4f7dc485f74109bd66ef74231285ff797a823/doc/README.md

> ```

Reviewer: I think that's just a [link reference definition](https://spec.commonmark.org/0.30/#link-reference-definitions), it doesn't appear in the rendered document

#### Inline comments

Inline comments are supported with following syntax:

```markdown
<!-- This is an inline comment -->
```

The comments will not be rendered in the rendered HTML.

#### Typographic replacements

The following [typographic replacements](https://github.com/executablebooks/markdown-it-py/blob/3613e8016ecafe21709471ee0032a90a4157c2d1/markdown_it/rules_core/replacements.py#L1-L15) are enabled:

* ``(c)``, ``(C)`` → ©
* ``(tm)``, ``(TM)`` → ™
* ``(r)``, ``(R)`` → ®
* ``+-`` → ±
* ``...`` → …
* ``?....`` → ?..
* ``!....`` → !..
* ``????????`` → ???
* ``!!!!!`` → !!!
* ``,,,`` → ,
* ``--`` → &ndash

> * ``---`` → &mdash

Reviewer: I'm wondering whether we need to document this at all, seems not very worthwhile, are any of these even used right now? Could we turn this extension off even and only enable it if it's actually needed?

Reviewer 2: We probably don't have to write them out explicitly as that would be maintenance burden, but adding a permalink is good IMO.
